analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/RoslynLiveVariableAnalysis.cs

```diff
@@ -51,94 +51,91 @@ public sealed class RoslynLiveVariableAnalysis : LiveVariableAnalysisBase<Contro
-                    case OperationKindEx.ParameterReference:
-                        ProcessParameterOrLocalReference(state, IParameterReferenceOperationWrapper.FromOperation(operation.Instance));
-                        break;
-                    case OperationKindEx.SimpleAssignment:
-                        ProcessSimpleAssignment(state, ISimpleAssignmentOperationWrapper.FromOperation(operation.Instance));
-                        break;
-                    case OperationKindEx.FlowAnonymousFunction:
-                        ProcessFlowAnonymousFunction(state, cfg, IFlowAnonymousFunctionOperationWrapper.FromOperation(operation.Instance));
-                        break;
+                    switch (operation.Instance.Kind)
+                    {
+                        case OperationKindEx.LocalReference:
+                            ProcessParameterOrLocalReference(ILocalReferenceOperationWrapper.FromOperation(operation.Instance));
+                            break;
+                        case OperationKindEx.ParameterReference:
+                            ProcessParameterOrLocalReference(IParameterReferenceOperationWrapper.FromOperation(operation.Instance));
+                            break;
+                        case OperationKindEx.SimpleAssignment:
+                            ProcessSimpleAssignment(ISimpleAssignmentOperationWrapper.FromOperation(operation.Instance));
+                            break;
+                        case OperationKindEx.FlowAnonymousFunction:
+                            ProcessFlowAnonymousFunction(cfg, IFlowAnonymousFunctionOperationWrapper.FromOperation(operation.Instance));
+                            break;
+                    }
                 }
             }
-        }
 
-        private void ProcessParameterOrLocalReference(State state, IOperationWrapper reference)
-        {
-            var symbol = ParameterOrLocalSymbol(reference.WrappedOperation);
-            Debug.Assert(symbol != null, "Only supported types should be passed to ParameterOrLocalSymbol");
-            if (IsOutArgument(reference.WrappedOperation))
+            private void ProcessParameterOrLocalReference(IOperationWrapper reference)
             {
-                state.Assigned.Add(symbol);
-                state.UsedBeforeAssigned.Remove(symbol);
-            }
-            else if (!IsAssignmentTarget())
-            {
-                state.UsedBeforeAssigned.Add(symbol);
-            }
-
-            bool IsAssignmentTarget() =>
-                new IOperationWrapperSonar(reference.WrappedOperation).Parent is { } parent
-                && parent.Kind == OperationKindEx.SimpleAssignment
-                && ISimpleAssignmentOperationWrapper.FromOperation(parent).Target == reference.WrappedOperation;
-        }
+                var symbol = ParameterOrLocalSymbol(reference.WrappedOperation);
+                Debug.Assert(symbol != null, "Only supported types should be passed to ParameterOrLocalSymbol");
```

Comment:
Minor: I think that it is better to have this assertion in the function you reference in the comment.

Reply:
It is normal for `ParameterOrLocalSymbol` to return null when invoked from `ProcessSimpleAssignment`. I'll try to improve the message.